src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -519,9 +529,24 @@ def _compile_vts(self, vts, sources, analysis_file, upstream_analysis, classpath
 
         # If compiling a plugin, don't try to use it on itself.
         javac_plugins_to_exclude = (t.plugin for t in vts.targets if isinstance(t, JavacPlugin))
-        self.compile(self._args, classpath, sources, outdir, upstream_analysis, analysis_file,
-                     log_file, settings, fatal_warnings, zinc_file_manager,
-                     javac_plugins_to_exclude)
+        try:
+          self.compile(self._args, classpath, sources, outdir, upstream_analysis, analysis_file,
+                       log_file, settings, fatal_warnings, zinc_file_manager,
+                       javac_plugins_to_exclude)
+        except TaskError:
+          def find_failed_compile_log(workunits):
```

Comment:
Defining this inside the `except` block makes this more challenging to read. Would prefer to see the function defined outside (class level, or otherwise) so that the `except` is small.